### PR TITLE
Optimise the fallback dispatcher by removing a useless jump

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

**Comment by axic** [comment]
Not sure yet why that single `external function requested` test fails with this change. Any ideas?

File: libsolidity/ast/Types.cpp [hunk 1]
```diff
@@ -2507,6 +2507,9 @@ string FunctionType::externalSignature() const
 {
 	solAssert(m_declaration != nullptr, "External signature of function needs declaration");
 
+	if (m_declaration->name().empty())
```
Comment (chriseth) [inline libsolidity/ast/Types.cpp]: Shouldn't it rather fail in this case?

Comment (axic) [reply libsolidity/ast/Types.cpp]: I use it to determine that it is a fallback. Should we just use `name().empty()` or rather introduce a helper `isFallback()`?

File: libsolidity/codegen/ContractCompiler.cpp [hunk 2]
```diff
@@ -533,7 +527,8 @@ bool ContractCompiler::visit(FunctionDefinition const& _function)
 
 	m_context.adjustStackOffset(-(int)c_returnValuesSize);
 
-	if (!_function.isConstructor())
+	/// The constructor and the fallback function doesn't to jump out.
+	if (!_function.isConstructor() && !_function.externalSignature().empty())
```
Comment (chriseth) [inline libsolidity/codegen/ContractCompiler.cpp]: Please use `_function.name().empty()` or introduce a helper function.

**Comment by axic** [comment]
@chriseth ready for review

**Review by chriseth — APPROVED** [review]

**Pull request merged**